### PR TITLE
#165250447 Article categories

### PR DESCRIPTION
**Pull request by Andhrah**
#### What does this PR do?
It creates article categories page

#### Description of task to be completed?
- write test for category page
- create category component
- build out category page

#### How should this be manually tested?
- Follow the steps below:
  - Run the following commands in your terminal
`git clone https://github.com/andela/apollo-ah-frontend.git`
`cd apollo-ah-frontend`
- Run `npm install` to install all dependencies.
- Start the app 
#### What are the relevant pivotal tracker stories?
[#165250447](https://www.pivotaltracker.com/story/show/165250447)
#### Screenshots (if appropriate)
articleCategory

#### Questions:

### CODE REVIEW

**Comment by fejiroofficial** [comment]
You have a host of conflicts to resolve @Andraquin.

File: src/selectors/categorySelector.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { createSelector } from 'reselect';
+
+/**
+ * @param {*} state - The initial state from the redux store
+ */
+const categorySelector = state => state.articlesCategoryReducer.category;
```
Comment (valenstical) [inline src/selectors/categorySelector.js]:
```suggestion
const categorySelector = state => state.categories;
```
Then in the combineReducer, we can have,
`{
  categories: articleCategoryReducer
}`

What do you think?

File: src/selectors/categorySelector.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+const categoryIsLoadingSelector = state => state.articlesReducer.loading;
```
Comment (valenstical) [inline src/selectors/categorySelector.js]:
```suggestion
const categoryIsLoadingSelector = state => state.articles.loading;
```
Similar to the suggestion above, we can have this in the combineReducer
`{
  articles: articlesReducer
}
`

File: src/styles/shared.scss [hunk 3]
```diff
@@ -198,7 +198,7 @@ body {
 	}
 }
 .wrapper {
-	padding-top: 50px;
+	padding-top: 100px;
```
Comment (valenstical) [inline src/styles/shared.scss]: _Hmmmm_, almost every page have this class _.wrapper_ assigned to them. Would adding an extra 50px to the top padding not affect the rest pages.

**Review by valenstical — APPROVED** [review]
LGTM

File: src/components/Category.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import { createStructuredSelector } from 'reselect';
+import FooterContainer from '../views/FooterContainer';
+import { getArticlesCategory } from '../actions';
+import PlaceholderLoader from '../views/placeholderLoader';
+import Category from '../views/Category';
+import * as categorySelector from '../selectors/categorySelector';
+
+
+/**
+ * @description - returns the home page view
```
Comment (aanchirinah) [inline src/components/Category.jsx]: #### Nitpick
- Does it though?

File: src/components/Category.jsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  render() {
+    const {
+      articlesCategory,
+      loadingCategory,
+    } = this.props;
+    return (
+      <section className="wrapper">
+        {loadingCategory === 'started' ? <PlaceholderLoader /> : <Category begin={0} end={12} articlesCategory={articlesCategory} />}
```
Comment (aanchirinah) [inline src/components/Category.jsx]: #### Nitpick
- DRY principle violation fix it.

File: src/reducers/articleReducer.js [hunk 6]
```diff
@@ -39,13 +39,13 @@ export default (state = initialState, action) => {
...
       };
-      case typeGenerator('GET_ARTICLES_FAILURE'):
+    case typeGenerator('GET_ARTICLES_FAILURE'):
       return {
         ...state,
         loading: action.payload,
         articles: 'Articles not found'
       };
-      case typeGenerator('SERVER_ERROR'):
+    case typeGenerator('SERVER_ERROR'):
```
Comment (aanchirinah) [inline src/reducers/articleReducer.js]: #### Nitpick
- Is this really necessary.

File: src/test/actions/articleCategoryAction.test.js [hunk 7]
```diff
@@ -9,34 +9,32 @@ const mockStore = configureMockStore(middlewares);
...
   });
   it('returns nothing if there is an error', () => {
-    mockAxios.get.mockImplementationOnce(()=> Promise.reject({}));
+    mockAxios.get.mockImplementationOnce(() => Promise.reject(new Error()));
 
     const expectedActions = [{
-      "payload": {},
-      "type": "SEVER_ERROR",
-      }];
+      payload: new Error(),
```
Comment (aanchirinah) [inline src/test/actions/articleCategoryAction.test.js]: #### Comment
- Hope this is not a case of editing a test to get yours to pass

File: src/test/components/Home.test.js [hunk 8]
```diff
@@ -1,7 +1,8 @@
+/* eslint-disable import/no-named-as-default */
 import React from 'react';
 import { configure, shallow } from 'enzyme';
 import Adapter from 'enzyme-adapter-react-16';
-import HomePage from '../../components/HomePage.jsx';
+import HomePage from '../../components/HomePage';
 
 configure({ adapter: new Adapter() });
```
Comment (aanchirinah) [inline src/test/components/Home.test.js]: #### Nitpick
- This can be configured to have one instance for all your tests.

**Review by aanchirinah — CHANGES_REQUESTED** [review]
I'm not particularly happy with the file changes regarding styles , it flags too many false postives.

**Comment by valenstical** [comment]
Hello, @Andraquin, what is the status of this PR now. Are we done with fixing the merge conflicts?

**Comment by Andhrah** [comment]
> Hello, @Andraquin, what is the status of this PR now. Are we done with fixing the merge conflicts?

@valenstical Yes I have.

**Review by aanchirinah — APPROVED** [review]

**Pull request merged**